### PR TITLE
Version 2.4.9

### PR DESCRIPTION
**Pull request by ioancatanoaiaosf**
- merge miniextensions into main module
- improved log information

### CODE REVIEW

**Comment by ioancatanoaiaosf** [comment]
@rachel-yankelevitz  we did this pull request containing some module improvements, please review.
Once approved we'll also start the process of releasing this new version on marketplace

File: Model/ActionsHandler/Decline.php [hunk 1]
```diff
@@ -143,10 +143,17 @@ private function cancelOrder($order)
         $order->cancel()->save();
         if ($order->isCanceled()) {
             $this->forterConfig->addCommentToOrder($order, 'Order Cancelled');
+            $this->forterConfig->log('Order Cancelled - orderID ' . $order->getIncrementId());
+            $this->forterConfig->log('Canceled Order '. $order->getIncrementId() . ' Data: ' . json_encode($order->getData()));
+            $this->forterConfig->log('Canceled Order '. $order->getIncrementId() . ' Payment Data: ' .json_encode($order->getPayment()->getData()));
```
Comment (pniel-cohen) [inline Model/ActionsHandler/Decline.php:148]: Why is that needed? (the data)

File: Model/ActionsHandler/Decline.php [hunk 2]
```diff
@@ -143,10 +143,17 @@ private function cancelOrder($order)
...
             return;
         }
 
         $this->forterConfig->addCommentToOrder($order, 'Order Cancellation attempt failed');
+        $this->forterConfig->log('Order Cancellation attempt failed - orderID ' . $order->getIncrementId());
+        $this->forterConfig->log('Cancellation Failure for Order ' . $order->getIncrementId() .' - Order Data: ' . json_encode($order->getData()));
+        $this->forterConfig->log('Cancellation Failure for Order ' . $order->getIncrementId() .' - Payment Data: ' . json_encode($order->getPayment()->getData()));
```
Comment (pniel-cohen) [inline Model/ActionsHandler/Decline.php:155]: Why is that needed? (the data).

File: Model/AbstractApi.php [hunk 3]
```diff
@@ -213,14 +242,307 @@ public function getUpdatedStatusEnum($order)
...
+                $this->json['payment'][0]['creditCard']['lastFourDigits'] = $paymentCheckoutCom['data']['source']['last_4'];
+            }
+            if ($paymentCheckoutCom) {
+                $this->json['payment'][0]['creditCard']['fullResponsePayload'] = $paymentCheckoutCom;
+            }
+            return;
+        } catch (\Exception $e) {
+            $this->reportToForterOnCatch($e);
+        }
+        return;
```
Comment (pniel-cohen) [inline Model/AbstractApi.php:545]: This should be refactored and live in a separate class that's meant for building schemas (like those that are under `Model/RequestBuilder/`). It shouldn't be part of the main AbstractApi.

File: etc/di.xml [hunk 4]
```diff
@@ -39,4 +39,9 @@
 			</argument>
 		</arguments>
 	</type>
+	<type name="Forter\Forter\Model\AbstractApi">
+		<arguments>
+			<argument name="serializer" xsi:type="object">Magento\Framework\Serialize\Serializer\Serialize</argument>
+		</arguments>
+	</type>
```
Comment (pniel-cohen) [inline etc/di.xml:46]: *See comments on AbstractApi changes.

**Review by pniel-cohen — DISMISSED** [review]
1. IMO, most of the new logs are not needed and making it look messy. The module already logs requests and responses and some other relevant stuff.
If there's a need to debug a unique issue on a specific site, additional logs may be added locally by the developer, on the specific environment and be removed afterward (it should not be committed to the main module).
2. Additional info for the `AbstractApi::sendOrderStatus` schema (unless it's a minor adjustment), should be separated into a dedicated RequestBuider class and not be added to the main AbstractApi class.

**Review by rachel-yankelevitz — APPROVED** [review]
lgtm
